kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala

```diff
@@ -144,10 +156,29 @@ class OperationLog(path: Path) {
       logs.addAll(extraLogs)
     }
 
-    val tColumn = TColumn.stringVal(new TStringColumn(logs, ByteBuffer.allocate(0)))
-    val tRow = new TRowSet(0, new JArrayList[TRow](logs.size()))
-    tRow.addToColumns(tColumn)
-    tRow
+    toRowSet(logs)
+  }
+
+  def read(from: Int, size: Int): TRowSet = synchronized {
+    var pos = from
+    if (pos < 0) {
+      // just fetch forward
+      pos = lastReadPos
```

Comment:
if pos <0, could we invoke `def read(maxRows: Int): TRowSet` directly?

Reply:
I see,
Now we have reader/extraReaders for read forward and seekableReader for read from offset.

And the lastReadPosition is not consistent.

Reply:
It seems not matter, only one more seekableReader will be created.

Reply:
yea, we should use the one reader to track the last position, otherwise it become complex